### PR TITLE
[1LP][RFR] all method filtering support NetworkProvider

### PR DESCRIPTION
**Pull request by digitronik**
Purpose or Intent
=================
- added support for filtering provider in `NetworkProvider` `all()` method
- related some changes. 
- may be tests releted to rhos11 will fail due to deletion problem going on with rhos provider. 

{{pytest: cfme/tests/networks/test_sdn_navigation.py  cfme/tests/networks/test_sdn_crud.py --use-provider complete -sqvv}}

### CODE REVIEW

**Comment by digitronik** [comment]
The failures are due to RHOS-11 deletion problem.

File: cfme/tests/networks/test_sdn_navigation.py [hunk 1]
```diff
@@ -21,11 +21,8 @@ def test_sdn_provider_relationships_navigation(provider, tested_part, appliance)
     Metadata:
         test_flag: sdn
     """
-    view = navigate_to(provider, 'Details')
-    net_prov_name = view.entities.summary("Relationships").get_text_of("Network Manager")
-
-    collection = appliance.collections.network_providers
-    network_provider = collection.instantiate(prov_class=NetworkProvider, name=net_prov_name)
+    collection = appliance.collections.network_providers.filter({'provider': provider})
+    network_provider = random.choice(collection.all())
```
Comment (psav) [inline cfme/tests/networks/test_sdn_navigation.py]: **REQUIRED**: Why using random here at all, wouldn't ```[0]``` do? Do providers generally have a lot of network providers?

Comment (digitronik) [reply cfme/tests/networks/test_sdn_navigation.py]: @psav Yes, I can go with`[0]`, but I thought sometimes we have two clouds providers(You can say to our bad or BZ) so I thought better to go with random.

Comment (digitronik) [reply cfme/tests/networks/test_sdn_navigation.py]: aaaaahhh understand :) It was my bad; actually I already filtering with the specific provider :+1:

**Comment by digitronik** [comment]
@psav: I added changes. Please review it onetime.

File: cfme/tests/networks/test_sdn_crud.py [hunk 2]
```diff
@@ -24,9 +23,8 @@ def test_sdn_crud(provider, appliance):
         test_flag: sdn
     """
     view = navigate_to(provider, 'Details')
-    net_prov_name = view.entities.summary("Relationships").get_text_of("Network Manager")
-    collection = appliance.collections.network_providers
-    network_provider = collection.instantiate(prov_class=NetworkProvider, name=net_prov_name)
+    collection = appliance.collections.network_providers.filter({'provider': provider})
```
Comment (mshriver) [inline cfme/tests/networks/test_sdn_crud.py]: The navigation above here is no longer necessary

**Pull request merged**